quiz/src/Category.js

```diff
@@ -0,0 +1,43 @@
+import React ,{useState,useEffect} from 'react';
+import {CategoryData,DifficultyData} from './Data';
+
+function Category () {
+  const[cat,setCat]=useState('');
+  const[diff,setDiff]=useState('');
+  const [questions,setQuestions]=useState([]);
+  const[show,setShow]=useState(true);
+  useEffect(()=>{
+
+    if(cat||diff){
+    fetch(`https://opentdb.com/api.php?amount=10&category=${cat}&difficulty=${diff}`)
+    .then(res => res.json()).then(({results})=>setQuestions(results))
+    .catch(error => console.error(error))}
+    else{
+      setQuestions([]);
+    }
```

Comment:
here there is no need to empty the array of questions , it is already empty.
**More explanation :** 

- `cat` & `diff` will be empty just at the first time " after the initial rendering " and at this time 
   questions array is empty so there is no need for `else statement` .
- Then during rendering the returned array of questions, you should check if the array is empty or not and show a descriptive message such as " No questions for this category " if the array is empty.